Edwin branch

EdwinZurawik opened the pull request:
Added basic player and enemy movement:
    
    -Added Actor, Player_Human, Player_Wasp and Enemy_Wasp scripts
    -Also added scenes for Player (Human, Wasp) and Enemy (Wasp)
    -World.gd added to Autoload as GlobalWorld
    -PlayerTypes added to World.gd
    -Created small test level using grass_tileset (it loads as a default
    level when you press F5)
    -Changed Project Preferences:
    a) rendering quality = pixel snap
    b) window>strech:  mode=2d, aspect=keep
    c) size: w=640,h=480 / test_w=1024,test_h=768
    -added src, Actors, Levels directories
    
    *I didn't clean up old Actor and Creature scenes!

vVendigo123 committed:
WIP Creatures

vVendigo123 committed:
Added basic player and enemy movement:

-Added Actor, Player_Human, Player_Wasp and Enemy_Wasp scripts
-Also added scenes for Player (Human, Wasp) and Enemy (Wasp)
-World.gd added to Autoload as GlobalWorld
-PlayerTypes added to World.gd
-Created small test level using grass_tileset (it loads as a default
level when you press F5)
-Changed Project Preferences:
a) rendering quality = pixel snap
b) window>strech:  mode=2d, aspect=keep
c) size: w=640,h=480 / test_w=1024,test_h=768
-added src, Actors, Levels directories

*I didn't clean up old Actor and Creature scenes!

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+
+func _ready() -> void:
+    self.speed.y = 300
+
+
```
Wystarczy jedna linia odstępu. Albo zero.

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+        $RayCast2D.position.x *= -1 
+        
+    return dir
+
+
```
To samo

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+    out.x = speed.x * dir.x
+    out.y += gravity * get_physics_process_delta_time()
+    return out
+    
+     
+
```
To samo

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+        dir: Vector2,
+        speed: Vector2
+    ) -> Vector2:
+       
+        
```
To samo

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+    out.x = speed.x * dir.x
+    out.y += gravity * get_physics_process_delta_time()
+    return out
+    
+     
+
```
Jedna linia za dużo

EdwinZurawik replied:
line removed

RcAzure commented:
Please remove old, unused classes/files and unused test level.

vVendigo123 committed:
removed unused files and scenes + removed line from Enemy_Wasp.gd

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+
+func get_direction(dir) -> Vector2:
+    if is_on_wall():
+        dir.x = dir.x * -1
```
nie wystarczy dir.x = -dir.x?

EdwinZurawik replied:
true, i'll change it right away

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+    if $RayCast2D.is_colliding() == false:
+        dir.x = dir.x * -1
+        $RayCast2D.position.x *= -1 
+        
```
Spróbuj połączyć te dwa ify w jeden za pomocą operatorów OR, żeby nie dublować tego samego kodu.

EdwinZurawik replied:
`func get_direction(dir) -> Vector2:
    if is_on_wall() or $RayCast2D.is_colliding() == false:
        dir.x = -dir.x
        $RayCast2D.position.x *= -1
        
    return dir`

EdwinZurawik replied:
```
func get_direction(dir) -> Vector2:
    if is_on_wall() or $RayCast2D.is_colliding() == false:
        dir.x = -dir.x
        $RayCast2D.position.x *= -1
        
    return dir
```

vVendigo123 committed:
Enemy_Wasp.gd - reduced code after suggestion

RcAzure commented on src/Actors/Enemy_Wasp.gd:
```diff
@@ -0,0 +1,46 @@
+     
+
+func flip_sprite(dir):
+    if dir.x < 0:
+        $AnimatedSprite.flip_h = true 
+    else: 
+        $AnimatedSprite.flip_h = false
```
a może AnimatedSprite.flip_h = dir.x < 0? Albo utwórz coś takiego: bool cond = dir.x < 0; $AnimatedSprite.flip_h = cond

EdwinZurawik replied:
```
func flip_sprite(dir):
        $AnimatedSprite.flip_h = true if dir.x < 0 else false 
```

RcAzure commented on src/Actors/Player_Human.gd:
```diff
@@ -0,0 +1,39 @@
+        out.y = 0.0
+    return out
+    
+func flip_sprite(direction):
```
To samo co w waspie. Przenieś to do Actora, jeśli to wspólna funkcjonalność.

EdwinZurawik replied:
done

RcAzure commented on src/Actors/Player_Wasp.gd:
```diff
@@ -0,0 +1,36 @@
+    out = speed * direction
+    return out
+
+func flip_sprite(direction):
+    
+    if direction.x < 0:
+        $AnimatedSprite.flip_h = true 
+    else: 
+        $AnimatedSprite.flip_h = false
+
```
Wspólna funkcjonalność

EdwinZurawik replied:
done

vVendigo123 committed:
Fixes after review:

-removed redundant if statement
-moved flip_sprite() to Actor.gd
-shortened flip_sprite() body to one line

RcAzure commented on src/Actors/Player_Wasp.gd:
```diff
@@ -29,8 +29,3 @@ func calculate_move_velocity(
 
 func flip_sprite(direction):
```
Została pusta funkcja

EdwinZurawik replied:
fck fixin

vVendigo123 committed:
removed empty function
